### PR TITLE
Classes and methods excercises

### PR DESCRIPTION
**Pull request by mlan080**
I am adding excercises all the time to self-learning-ninja file mainly from the The Ruby programming language book which is quite challenging to read through ind detail and understand everything but I am focusing on Alex's tasks he has outlined.

### CODE REVIEW

File: 6. Objects and Methods /attr_reader.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Point
+  #shorter version
+  attr_reader :x, :y  #define accessor methods for our instance variables so users of the Point class can use it
+
+  def initialize(x,y)
+    @x, @y = x,y
+  end
+
+  def to_s #return a string that represents this point
+    "(#@x, #@y)" #interpolate the instance variables into the string
```
Comment (mlan080) [inline 6. Objects and Methods /attr_reader.rb:10]: So if I use string interpolation principles I can get rid of 9,10,11 and just make line 25 into "#{p}" however this doesn't return (1,2) it returns the instance variable. Can I have some guidance?

Comment (lucasrenan) [reply 6. Objects and Methods /attr_reader.rb:10]: the `to_s` method is a great practice. 

giving an instance of `Point`, when printing it, the `to_s` method is called.

the problem here is the interpolation, it should be:
`"(#{x}, #{y})"`

File: 6. Objects and Methods /instance_variable_example.rb [hunk 2]
```diff
@@ -6,17 +6,16 @@ def initialize(id, name, addr)
...
    end
-   def display_details()
-      puts "Customer id #@cust_id"
-      puts "Customer name #@cust_name"
-      puts "Customer address #@cust_addr"
-   end
-end
+
+   def to_s
+    "(#{@cust_id}, #{@cust_id}, #{@cust_id})"
```
Comment (mlan080) [inline 6. Objects and Methods /instance_variable_example.rb:11]: As per our string interpolation logic, without the @,  why is there is an error `to_s': undefined local variable or method `cust_id' for #<Customer:0x00007ffb1e9e2b88> (NameError)

Comment (lucasrenan) [reply 6. Objects and Methods /instance_variable_example.rb:11]: because there is no attr reader for `cust_id`

**Pull request merged**